src/containers/IdentityContainer.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from 'react';
+
+import { esriPromise } from 'react-arcgis';
+
+import { Identity } from '../components/Identity';
+
+interface IdentityComponentProps {
+  credential?: __esri.Credential;
+  portalUrl: string;
+  handleLoginChange: (credential?: __esri.Credential) => void;
+}
+interface IdentityComponentState {}
+
+export class IdentityContainer extends React.Component<IdentityComponentProps, IdentityComponentState> {
+  identityManager: __esri.IdentityManager;
+
+  constructor(props: IdentityComponentProps) {
+    super(props);
+    this.handleLogin = this.handleLogin.bind(this);
+    this.handleLogout = this.handleLogout.bind(this);
+  }
+
+  componentDidMount() {
+    esriPromise([
+      'esri/identity/OAuthInfo',
+      'esri/identity/IdentityManager'
+    ]).then(([OAuthInfo, IdentityManager]) => {
+      this.identityManager = IdentityManager;
+
+      const info = new OAuthInfo({
+          appId: 'y4Lx1l6456Mbf85z',
+          popup: false
+      });
+
+      this.identityManager.registerOAuthInfos([info]);
+      this.identityManager.checkSignInStatus(this.props.portalUrl).then(
+        (credential: __esri.Credential) => {
+          this.props.handleLoginChange(credential);
+        }
+      );
+    });
+  }
+
+  handleLogout() {
+    if (this.identityManager) {
+      this.identityManager.destroyCredentials();
+      this.props.handleLoginChange();
```

Comment:
While manually calling `this.props.handleLoginChange()` when someone presses the `Logout` button works, I think a more robust solution is to subscribe to [the events emitted by the `IdentityManager`](https://developers.arcgis.com/javascript/latest/api-reference/esri-identity-IdentityManager.html#events-summary) when we mount the component.  That way, if we someday add some other tooling that somehow logs the user in/out, we don't have to think about manually calling that `props.handleLoginChange` method.

I think tying in to an object's events on mount is the best practice for keeping a non-React.js-based object in sync with your React app.  I found [this blog post](https://blog.mapbox.com/mapbox-gl-js-react-764da6cc074a) helpful (in particular, check out the `componentDidMount` of these examples: [visualizing lat/lng/zoom](https://jsfiddle.net/tristen/r3zg55tw), [tooltips on hover](https://jsfiddle.net/tristen/2p3fsgnu/)).  Whenever a third-party tool has events, we should try to make use of them to simplifying syncing our app's state with the state of the third-party objects.

Reply:
OK, I changed that. 

Had to introduce `interface IdentityManagerEvent` to please TypeScript. Turns out, that the event handler is passed not a `Credential` object but an `Object` with a property `credential`, which is a `Credential` object. To access that property and make TypeScript compile, I need to tell TypeScript the structure of the object.

Don't know if `IdentityContainer` is the right place to put it or if we should open `src/@types` for custom types? Guess we have to do some refactoring anyway if we decide to move forward with React so we could tackle it then.